src/mixer/basetrackplayer.cpp

```diff
@@ -216,16 +216,29 @@ void BaseTrackPlayerImpl::loadTrack(TrackPointer pTrack) {
     if (!m_pChannelToCloneFrom) {
         const QList<CuePointer> trackCues(m_pLoadedTrack->getCuePoints());
         QListIterator<CuePointer> it(trackCues);
+        CuePointer pLoopCue;
+        // Restore loop from the first loop cue with minimum hotcue number.
+        // For the volatile "most recent loop" the hotcue number will be -1.
+        // If no such loop exists, restore a saved loop cue.
         while (it.hasNext()) {
             CuePointer pCue(it.next());
-            if (pCue->getType() == mixxx::CueType::Loop) {
-                double loopStart = pCue->getPosition();
-                double loopEnd = loopStart + pCue->getLength();
-                if (loopStart != kNoTrigger && loopEnd != kNoTrigger && loopStart <= loopEnd) {
-                    m_pLoopInPoint->set(loopStart);
-                    m_pLoopOutPoint->set(loopEnd);
-                    break;
-                }
+            if (pCue->getType() != mixxx::CueType::Loop) {
+                continue;
+            }
+
+            if (pLoopCue && pLoopCue->getHotCue() <= pCue->getHotCue()) {
```

Comment:
Should this be `<` rather than `<=`? No two cues should have the same hotcue number.

Reply:
Two loop should not have the same hotcue number, but in case they do I wanted to make sure that the *first* loop is used, just like when unloading the track.

Reply:
This is the perfect use case for adding a code comment that is visible to everyone instead of giving the answer here ;)

Reply:
Already did: https://github.com/mixxxdj/mixxx/pull/2678/files/7541fe5f2cb07c6a39cc0e188938dddb36c70715#diff-4e68d45227023551fba42d24a56eed76R220

Reply:
I wan't expecting anything else from you, Jan ;)

Reply:
Please move the comment from above the loop to above this line.

Reply:
Nevermind, it's fine how it is.